inc/common.class.php

```diff
@@ -211,4 +211,192 @@ static function cloneItem(Array $param) {
+                                'behaviors');
+            }
+         }
+
+         if ($config->getField('is_tickettechgroup_mandatory')) {
+            if (($obj->countGroups(CommonITILActor::ASSIGN) == 0)) {
+
+               $warnings[] = __("Group of technicians assigned is mandatory before ticket is solved/closed",
+                                'behaviors');
+            }
+         }
+
+         if ($config->getField('is_ticketlocation_mandatory')) {
+            if ($loc == 0) {
+               $warnings[] = __("Location is mandatory before ticket is solved/closed", 'behaviors');
+            }
+         }
+
+         if ($config->getField('is_tickettasktodo')) {
+            foreach ($DB->request('glpi_tickettasks',
+                                 ['tickets_id' => $obj->getField('id')]) as $task) {
+               if ($task['state'] == 1) {
+                  $warnings[] = __("You cannot solve/close a ticket with task do to", 'behaviors');
+                  break;
+               }
+            }
+         }
+      }
+
+      if ($obj->getType() == 'Problem') {
+         if ($config->getField('is_problemtasktodo')) {
+            foreach ($DB->request('glpi_problemtasks',
+                                 ['problems_id' => $obj->getField('id')]) as $task) {
+               if ($task['state'] == 1) {
+                  $warnings[] = __("You cannot solve/close a problem with task do to", 'behaviors');
+                  break;
+               }
+            }
+         }
+      }
+
+      if ($obj->getType() == 'Change') {
+         if ($config->getField('is_changetasktodo')) {
+            foreach ($DB->request('glpi_changetasks',
+                                 ['changes_id' => $obj->getField('id')]) as $task) {
+               if ($task['state'] == 1) {
+                  $warnings[] = __("You cannot solve/close a change with task do to", 'behaviors');
+                  break;
+               }
+            }
+         }
+      }
+      return $warnings;
+   }
+
+
+   /**
+    * Displaying message solution
+    *
+    * @param $params
+   **/
+   static function messageWarningTicket($params) {
+
+      if (isset($params['item'])) {
+         $item = $params['item'];
+         if ($item->getType() == 'ITILSolution') {
+            $warnings = self::checkWarnings($params);
+            if (is_array($warnings) && count($warnings)) {
+               echo "<div class='alert alert-warning'>";
+
+               echo "<div class='d-flex'>";
+
+               echo "<div class='me-2'>";
+               echo "<i class='fa fa-exclamation-triangle fa-2x'></i>";
+               echo "</div>";
+
+               echo "<div>";
+               echo "<h4 class='alert-title'>" . __('You cannot resolve the ticket', 'behaviors') . "</h4>";
+               echo "<div class='text-muted'>" . implode('</div><div>', $warnings) . "</div>";
+               echo "</div>";
```

Comment:
missing part for ticket solutiontype 
(i can't resolve the conflict)

Reply:
I don't understand why thuis part has missing ? it's ok for me. For conflict, i can revert changes for inc/itilsolution.class.php temporarily ?